src/main/scala/scala/slick/jdbc/TypedStaticQuery.scala

```diff
@@ -0,0 +1,396 @@
+package scala.slick.jdbc
+
+import com.typesafe.config.{ ConfigFactory, ConfigException }
+import scala.annotation.{Annotation, StaticAnnotation}
+import scala.language.experimental.macros
+import scala.reflect.ClassTag
+import scala.reflect.macros.Context
+import scala.slick.collection.heterogenous._
+
+/**
+ * An implementation of the macros involved in the Plain SQL API
+ */
+object TypedStaticQuery {
+  
+  /**
+   * An annotation used with the tsql interpolation macro for defining
+   * the configuration of compile-time database connections through a config file
+   */
+  final class TSQLConfig(val dbName: String) extends Annotation with StaticAnnotation
+
+  /**
+   * The function used to fetch a ConfigHandler instance that ensures
+   * uniform database connections at compile-time and at run-time.
+   * tsql interpolation macro must always be used in conjunction with a
+   * ConfigHandler, more specifically a ConfigHandler.connection
+   */
+  def getConfigHandler() = macro getCHimpl
```

Comment:
The return type is statically known, right? It should be declared here, otherwise IntelliJ doesn't see it